### PR TITLE
Correct model layer aliasing for Bert style models

### PR DESCRIPTION
**Pull request by ap9272**
Correct model layer aliasing for Bert and ModernBert style models.

## Purpose

When loading Hugging Face checkpoints for BERT-style models (specifically `ModernBertModel` and `BertModel`), weight loading failed due to mismatched parameter names (aliasing) or unexpected task-specific weights in the checkpoint (e.g., Masked Language Modeling heads) that are not defined in the vLLM backbone model.

This PR resolves these issues:
1. **ModernBertModel (`vllm/model_executor/models/modernbert.py`)**:
   - Updated `hf_to_vllm_mapper` to properly map `model.layers.` and `layers.` to `encoder_layer.layers.`, and strip `model.` prefix.
   - Added a safeguard in `load_weights` to skip any weights not found in `params_dict` (e.g., extra MLM heads like `cls.predictions.*` or sequence classification heads in checkpoint files). This prevents a `KeyError` crash during model initialization.
3. **Warmup issues (`vllm/model_executor/warmup/deep_gemm_warmup.py`)**:
    - Moved `get_mk_alignment_for_contiguous_layout()` call to after the check for FP8. This would fail for model warmup on platforms that don't support DeepGEMM  without FP8 linearization (like TPUs)

## Test Plan

Ran the following models on GPU (Solves #28564 )
 - answerdotai/ModernBERT-base
 - answerdotai/ModernBERT-large

Ran the following models through tpu-inference on TPUs (relevant PR: [Encoder only attention support](https://github.com/vllm-project/tpu-inference/pull/2753)):
 - answerdotai/ModernBERT-base
 - answerdotai/ModernBERT-large
 - nvidia/nemocurator-fineweb-nemotron-4-edu-classifier
 - nvidia/nemocurator-fineweb-mixtral-edu-classifier
 - Alibaba-NLP/gte-modernbert-base

### CODE REVIEW

File: vllm/model_executor/warmup/deep_gemm_warmup.py [hunk 1]
```diff
@@ -147,6 +144,8 @@ def _fp8_linear_may_use_deep_gemm(module: torch.nn.Module) -> bool:
     ):
         return False
 
+    block_size = get_mk_alignment_for_contiguous_layout()[0]
+
```
Comment (mgoin) [inline vllm/model_executor/warmup/deep_gemm_warmup.py:157]: This change isn't mentioned in your description, what is it for?

**Comment by mgoin** [comment]
I see in our testing for this architecture we use `Alibaba-NLP/gte-modernbert-base`
https://github.com/vllm-project/vllm/blob/658b54efe419d0e53ec33a8bb7095d8c8b52c741/tests/models/registry.py#L603-L605
Does that checkpoint not have the problem you ran into?

**Comment by mgoin** [comment]
****

File: vllm/model_executor/warmup/deep_gemm_warmup.py (hunk 1 above)
Comment (ap9272) [reply vllm/model_executor/warmup/deep_gemm_warmup.py:157]: Missed it. Updated the description.
Basically, this call would fail if DeepGEMM without FP8 was not supported but the check was after this call.

**Comment by ap9272** [comment]
> I see in our testing for this architecture we use `Alibaba-NLP/gte-modernbert-base`
> 
> https://github.com/vllm-project/vllm/blob/658b54efe419d0e53ec33a8bb7095d8c8b52c741/tests/models/registry.py#L603-L605
> 
> 
> Does that checkpoint not have the problem you ran into?

I ran into the same issue that I've linked in the description.

**Review by mgoin — APPROVED** [review]
